pyvista/core/filters/poly_data.py

```diff
@@ -3915,8 +3915,9 @@ def voxelize_binary_mask(
     ):
         """Voxelize :class:`~pyvista.PolyData` as a binary :class:`~pyvista.ImageData` mask.
 
-        Generate a voxelized representation of a surface. The voxelization can be
-        controlled in several ways.
+        Generate a voxelized representation of a surface where the voxels enclosed by the
+        surface are referred to as foreground. The voxelization can be controlled in several
+        ways.
```

Comment:
Should we describe how a point and/or a voxel is considered background or foreground?

Reply:
> Should we describe how a point and/or a voxel is considered background or foreground?

Probably a good idea given the new examples, which are interesting cases that raise questions about how the filter works.

```
Generate a voxelized representation of a surface where the voxels enclosed by the
surface are referred to as foreground. The voxelization can be controlled in several
ways.
```

Might be good to include more details about what is meant by "enclosed". The specifics on how the algorithm works are detailed in a [source code comment](https://github.com/Kitware/VTK/blob/5fd6ec84fee1af9c4171fdee2a107ea0e49f5f1b/Imaging/Stencil/vtkPolyDataToImageStencil.cxx#L441-L448):

>> // Description of algorithm:
>>  // 1) cut the polydata at each z slice to create polylines
>>  // 2) find all "loose ends" and connect them to make polygons
>>  //    (if the input polydata is closed, there will be no loose ends)
>>  // 3) go through all line segments, and for each integer y value on
>>  //    a line segment, store the x value at that point in a bucket
>>  // 4) for each z integer index, find all the stored x values
>>  //    and use them to create one z slice of the vtkStencilData

There are also helpful comments in the implementation. Not sure what level of detail to include...

Reply:
See the proposed description.